### PR TITLE
👩🏼‍💻 Containerise solution

### LINKED ISSUE #506: 👩🏼‍💻 Containerise Solution
Cc: @GordonBeeming 
Hey @matt-goldman, 

We briefly discussed containerisation of SSW.Rewards, and I think it would be awesome to get that up and running!

# The Pain:

SSW Rewards requires some effort to set up and run properly. It would be a much better F5 Experience if we could simply run `docker compose build` and `docker compose up` to run our entire back end and get developing!

## Solution:

- [x] See the existing `./.docs/instructions_compile.md`

## Set up containers:
- [x] [Azurite - Default Ports](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=docker-hub,blob-storage)
- [x] [SQL Server - DB](https://learn.microsoft.com/en-us/sql/linux/quickstart-install-connect-docker?view=sql-server-ver16&pivots=cs1-bash)
- [x] `WebAPI.csproj`
- [x]  `AdminUI.csproj`
- [x] Https for AdminUI in Container. Using `dotnet run` and Dev Certs

Gold plating
- [ ] Automate Dev Tunnels for WebAPI(?)
- [ ] [SSW.Identity](https://github.com/SSWConsulting/SSW.IdentityServer) (Nice to have running local) A PBI Exists to have Identity Support Localhost on dev https://github.com/SSWConsulting/SSW.IdentityServer/issues/172
- [ ] Other services that would be great to run locally

1. Do we still need DevTunnels if we are running API on Docker?

### PR DESCRIPTION
**Pull request by tkapa**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ Closes #506 

> 2. What was changed?

✏️ Adds `Dockerfiles` and `docker-compose.yml` in order to run the WebApi and Admin Portal in Docker. 

The Docker Compose includes 
- Azure SQL Edge
- Azurite
- WebAPI
- AdminUI

Projects will now point to Identity Staging.

F5 Experience overhauled.

> 3. Did you do pair or mob programming?

✏️ TODO: Tested by the Rewards team

### CODE REVIEW

**Review by matt-goldman — APPROVED** [review]

File: _docs/Instructions-Compile.md [hunk 1]
```diff
@@ -1,94 +1,97 @@
...
-10. Copy over all files in `SeedData/profiles` (images + .xlsx) into the newly created `profile` container
-11. Press f5
-12. Test that you can access the swagger docs @ `https://localhost:5001/swagger/`
-
-**(Optional)** - If you need to run the api and mobile app locally. 
-1. Run dev tunnels `devtunnel host -p 5001`
+
+## Setting up the Repo for Development
+### To work on the API + Admin UI
+1. Pull this Repo
```
Comment (GordonBeeming) [inline _docs/Instructions-Compile.md:25]: @tkapa this should be "Clone this Repo"

File: _docs/Instructions-Compile.md [hunk 2]
```diff
@@ -1,94 +1,97 @@
...
+2. Grab the Secrets from Keeper 
+   1. **Client Secrets | SSW | SSW.Rewards | Developer Secrets**
+   2. Add them as .NET User Secrets for `WebAPI.csproj`
+   3. Update `appsettings.*.json` files accordingly
```
Comment (GordonBeeming) [inline _docs/Instructions-Compile.md:29]: This is ambiguous... what does update them accordingly mean?

Is there specific settings that come from keeper? Maybe explain this a little more, or create another md with deets if it's complicated

File: _docs/Instructions-Compile.md [hunk 3]
```diff
@@ -1,94 +1,97 @@
...
+3. Create a Developer Certificate https://learn.microsoft.com/en-us/aspnet/core/security/docker-https?view=aspnetcore-8.0#certificates
```
Comment (GordonBeeming) [inline _docs/Instructions-Compile.md:30]: Can this not be done as part of an `up.ps1` script?

Comment (GordonBeeming) [reply _docs/Instructions-Compile.md:30]: That would replace the next couple of steps

Comment (GordonBeeming) [reply _docs/Instructions-Compile.md:30]: Take a look at the coding challenge repo as an example 

https://github.com/SSWConsulting/SSW.CodingChallengeReimagined/blob/main/up.ps1

File: _docs/Instructions-Compile.md (hunk 1 above)
Comment (tkapa) [reply _docs/Instructions-Compile.md:25]: Hey @GordonBeeming, I agree and have updated it to match your wording.

File: _docs/Instructions-Compile.md (hunk 2 above)
Comment (tkapa) [reply _docs/Instructions-Compile.md:29]: Hey @GordonBeeming,

I've updated this, please see #589

File: _docs/Instructions-Compile.md (hunk 3 above)
Comment (tkapa) [reply _docs/Instructions-Compile.md:30]: @GordonBeeming,

Great suggestion, I've created a new issue https://github.com/SSWConsulting/SSW.Rewards.Mobile/issues/588

**Pull request merged**